src/Stays/StaysTypes.ts

```diff
@@ -53,6 +53,25 @@ export interface StaysRateCancellationTimeline {
   currency: string
 }
 
+export type StaysLoyaltyProgramme =
```

Comment:
one small things I'd change is keeping null separate in case we want to use `StaysLoyaltyProgramme` as a type for when we have certainty it will be present (so we can avoid doing the null checks over and over).

Reply:
Good point, done!